fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/ImmediateLocalFileReader.java

```diff
@@ -0,0 +1,562 @@
+            lock.lock();
+            try {
+                inflight--;
+                logger.info("Job completed inflight count now: " + inflight);
+
+                // Apply some hysteresis before we wake up the allocation
+                // thread. This is so we fetch larger allocations in fewer
+                // calls...although this isn't strictly necessary because
+                // here we are reading from a prepared list.
+                if (inflight < rescanThreshold) {
+                    logger.info("triggering job fetch [local file]");
+                    resourceLimit.signal();
+                }
+            } finally {
+                lock.unlock();
+            }
+        }
+    }
+
+    /**
+     * Submit the job to the internal threadpool
+     * @param job
+     */
+    private void process(BucketLoaderJob job) {
+        pool.submit(() -> {
+            processThr(job);
+        });
+    }
+
+    /**
+     * Process this job in a thread-pool thread
+     * @param job
+     */
+    private void processThr(final BucketLoaderJob job) {
+        // Ask the COS client to open the stream for the object and pass it to
+        // our process method for reading
+        try {
+            logger.info("Processing job: " + job.toString());
+            job.setProcessingStartTime(System.nanoTime());
+            switch (job.getFileType()) {
+            case NDJSON:
+                // Process the object using our NDJSON reader
+                readFile(job, br -> processNDJSON(job, br));
+                break;
+            case JSON:
+                // Process the object using our JSON reader
+                readFile(job, rdr -> processJSON(job, rdr));
+                break;
+            default:
+                logger.warning("Unrecognized file type for job: " + job.toString());
+                break;
+            }
+        } catch (Exception x) {
+            // make sure we don't propagate exceptions back to the pool thread. This probably
+            // means we hit a permission issue when scanning
+            logger.log(Level.SEVERE, "Error processing job: " + job.toString(), x);
+        }
+    }
+
+    /**
+     * Read the file described by the job and feed the contents to the given consumer
+     * @param job
+     * @param consumer
+     * @throws IOException
+     */
+    private void readFile(final BucketLoaderJob job, Consumer<BufferedReader> consumer) throws IOException {
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(job.getObjectKey()), StandardCharsets.UTF_8))) {
+            consumer.accept(br);
+        }
+    }
+
+    /**
+     * Process a JSON stream (as opposed to an NDJSON stream)
+     * @param job
+     * @param reader
+     */
+    private void processJSON(final BucketLoaderJob job, final Reader reader) {
+        final int lineNumber = 0;
+        try {
+            process(job, FHIRParser.parser(Format.JSON).parse(reader), lineNumber, "");
```

Comment:
wondering if we should have an overloaded process method...feels a bit odd to pass lineNumber=0 and line=""

or maybe just add a comment (either here or in javadoc for process() )?

Reply:
added some javadoc